gi-gtk-declarative/src/GI/Gtk/Declarative/Container/Patch.hs

```diff
@@ -76,6 +76,7 @@ patchInContainer (StateTreeContainer top children) container os' ns' = do
     -- widget in the corresponding place, we need to replace the GTK widget with
     -- one created from the declarative widget.
     (i, Just oldChildState, Nothing, Just new) -> do
+      error "this shouldn't happen!"
```

Comment:
I would like to call `destroy` here with `oldChildState`, but we can't do that because `destroy` also requires the old declarative widget - which we don't have.

Does this situation ever actually happen? It seems to me that if `oldChildState` exists then we should also have the old declarative widget (since that would have been used to create `oldChildState`).